### PR TITLE
Apollo module

### PR DESCRIPTION
**Pull request by sara-02**
Depends on #117  to be merged first

### CODE REVIEW

**Comment by sara-02** [comment]
For the user story https://github.com/openshiftio/openshift.io/issues/1286

**Comment by msrb** [comment]
[test]

**Comment by pkajaba** [comment]
@sara-02 please, do the rebase and I'll do the review afterward.

**Comment by sara-02** [comment]
@pkajaba Will update this after the reviews fro #117 are complete.

File: analytics_platform/kronos/apollo/src/apollo_tag_prune.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                package_tag_count = Counter()
+                for tag in tag_list:
+                    package_tag_count[tag] = word_count[tag]
+                tag_list = [package_tag[0].lower()
+                            for package_tag in package_tag_count.most_common(4)]
+            pruned_package_list[package_name] = tag_list
+        return pruned_package_list
+
+    @classmethod
+    def clean_each_file(cls, package_file,
```
Comment (jpopelka) [inline analytics_platform/kronos/apollo/src/apollo_tag_prune.py]: - Why is the method called clean_**each**_file() when it handles only one file ?
Either rename it to `clean_file()` or make it accept an iterable (a list of package file names for example) and then iterate over it in the method.

- You can change `@classmethod` to `@staticmethod` and remove the `cls` argument.

File: analytics_platform/kronos/apollo/src/apollo_tag_prune.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+           :param output_data_store: The output data store where clean package_topics are saved."""
+
+        package_topic_formatter_obj = TagListPruner(result_package_topic_json)
+        output_filename = additional_path + \
+            PACKAGE_LIST_OUTPUT_FILEPATH + package_file.split("/")[-1]
+        package_topic_formatter_obj.save(
+            output_package_topic_data_store, output_filename)
+
+    @classmethod
+    def prune_tag_list_max_count(cls, package_list):
```
Comment (jpopelka) [inline analytics_platform/kronos/apollo/src/apollo_tag_prune.py]: You don't use the `cls` inside of the method, so you can change `@classmethod` to `@staticmethod` and remove the `cls` argument.

File: tests/unit_tests/test_apollo_tag_prune.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        output_data_store = LocalFileSystem(
+            "tests/data/data_apollo/")
+        self.assertTrue(output_data_store is not None)
+
+        TagListPruner.get_prune_list(input_data_store,
+                                     output_data_store,
+                                     additional_path="")
+
+        output_pruned_list_obj = TagListPruner.load(data_store=output_data_store,
+                                                    filename="data_input_curated_package_topic/package_topic.json")
```
Comment (jpopelka) [inline tests/unit_tests/test_apollo_tag_prune.py]: Could you take the *"data_input_curated_package_topic"* from `apollo_constants.py` ?

File: tests/unit_tests/test_apollo_tag_prune.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        self.assertTrue(output_pruned_list_obj is not None)
+        output_result = output_pruned_list_obj.package_list
+        self.assertTrue(output_result is not None)
+
+        expected_pruned_list_obj = TagListPruner.load(data_store=output_data_store,
+                                                      filename="data_input_curated_package_topic/package_topic.json")
```
Comment (jpopelka) [inline tests/unit_tests/test_apollo_tag_prune.py]: I guess you want to load the `expected_output.json` here.

File: tests/data/data_apollo/data_input_curated_package_topic/package_topic.json [hunk 5]
```diff
@@ -0,0 +1 @@
+[{"ecosystem": "pypi", "package_topic_map": {"service_identity": ["security", "indentity", "cryptography", "plugin"], "pytest-pep8": ["pep8", "pytest", "plugin"]}}, {"ecosystem": "maven", "package_topic_map": {"org.pac4j:pac4j-mongo": ["authentication", "mongo", "secure"], "org.apache.mina:mina-core": ["network", "socket"], "org.bouncycastle:bcpkix-jdk15on": ["encryption", "cryptography"], "backports.ssl-match-hostname": ["secure", "cryptography", "encryption", "server"]}}, {"ecosystem": "npm", "package_topic_map": {}}, {"ecosystem": "ruby", "package_topic_map": {"service_identity": ["service_identity"]}}]
```
Comment (jpopelka) [inline tests/data/data_apollo/data_input_curated_package_topic/package_topic.json]: I think this file was added by mistake ?

**Comment by sara-02** [comment]
@jpopelka No, the work is complete, apart the suggested reviews.

File: analytics_platform/kronos/apollo/src/apollo_constants.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+"""Constants for Manifest Formating."""
+
+APOLLO_ECOSYSTEM = "ecosystem"
+APOLLO_INPUT_PATH = "data_input_raw_package_list/"
```
[the remark this answers is not in the document]
Comment (sara-02) [reply analytics_platform/kronos/apollo/src/apollo_constants.py]: @jpopelka Yes, thanks for pointing this out.

File: analytics_platform/kronos/apollo/src/apollo_constants.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+APOLLO_PACKAGE_LIST = "package_topic_map"
+
+"""Constants for output Manifest file."""
+
+PACKAGE_LIST_OUTPUT_FILEPATH = "data_input_curated_package_topic/"
```
[the remark this answers is not in the document]
Comment (sara-02) [reply analytics_platform/kronos/apollo/src/apollo_constants.py]: +1

File: analytics_platform/kronos/apollo/src/apollo_tag_prune.py [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+                       output_package_topic_data_store,
+                       additional_path):
+        """Generate the clean aggregated package_topic list as required by Gnosis.
+
+        :param input_package_topic_data_store: The Data store to pick the package_topic files from.
+        :param output_package_topic_data_store: The Data store to save the clean package_topic to.
+        :param additional_path: The directory to pick the package_topic files from."""
+
+        package_list_files = input_package_topic_data_store.list_files(
+            additional_path + APOLLO_INPUT_PATH)
```
[the remark this answers is not in the document]
Comment (sara-02) [reply analytics_platform/kronos/apollo/src/apollo_tag_prune.py]: Will to make similar checks across other modules as well, will create a separate issue for it.

File: tests/data/data_apollo/data_input_curated_package_topic/package_topic.json (hunk 5 above)
Comment (sara-02) [reply tests/data/data_apollo/data_input_curated_package_topic/package_topic.json]: Nopes, it is the data file used by the `test_apollo_tag_prune` file.

Comment (sara-02) [reply tests/data/data_apollo/data_input_curated_package_topic/package_topic.json]: @jpopelka My bad, yes it is an extra file. Will remove it.

File: tests/unit_tests/test_apollo_tag_prune.py (hunk 4 above)
Comment (sara-02) [reply tests/unit_tests/test_apollo_tag_prune.py]: Thanks for pointing this out.

**Comment by sara-02** [comment]
@jpopelka  PTAL

File: tests/data/data_apollo/data_input_curated_package_topic/package_topic.json (hunk 5 above)
Comment (jpopelka) [reply tests/data/data_apollo/data_input_curated_package_topic/package_topic.json]: Yes, but the test AFAICT generates it, so it does not need to be in git. But doesn't do any harm, so I'm fine with that.

**Pull request merged**